tools/mo/openvino/tools/mo/moc_frontend/pipeline.py

```diff
@@ -136,17 +136,48 @@ def create_target_input_shapes(new_input_places):
 
     if freeze_placeholder:
         for name, value in freeze_placeholder.items():
-            for node in user_shapes:
-                if node.get('input_name') == name:
-                    place = node['node']
-                    if node.get('shape'):
-                        input_model.set_partial_shape(place, node['shape'])
-                    if node.get('data_type'):
-                        value = np.array(value, dtype=node['data_type'])
-                        input_model.set_element_type(place, Type(node['data_type']))
-                    else:
-                        value = np.array(value, dtype=np.float32)
-                    input_model.set_tensor_value(place, value)
+            node = None
+            # look for the certain place in user_shapes
+            for node_cur in user_shapes:
+                if node_cur.get('input_name') == name:
+                    node = node_cur
+                    break
+            assert node is not None, "Please check correctness of the command-line. " \
+                                     "Place (operation or tensor) with name {} is not found.".format(name)
+            place = node.get('node')
+
+            if node.get('shape'):
+                input_model.set_partial_shape(place, node['shape'])
+
+            if node.get('data_type'):
+                dtype = node['data_type']
+            else:
+                # we need to detect type of Placeholder
+                try:
+                    ov_type = input_model.get_element_type(place)
+                except NotImplementedFailure:
+                    raise Error("Please specify type for value freezing {} node explicitly "
+                                "because the frontend does not support automatic type detection.".format(name))
+                # in case of cutting graph (or using custom inputs) and unspecified type,
+                # the default type is fp32
+                if ov_type == Type.undefined:
+                    ov_type = Type.f32
+                dtype = get_numpy_ctype(ov_type)
+
+            input_model.set_element_type(place, Type(dtype))
```

Comment:
In this particular case is it necessary to obtain ov_type, convert to numpy and then back to `Type`?

Reply:
Fixed.